x-pack/legacy/server/lib/register_license_checker/register_license_checker.js

```diff
@@ -4,21 +4,27 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { mirrorPluginStatus } from '../mirror_plugin_status';
+import { pairwise } from 'rxjs/operators';
+
+import { ServiceStatusLevels } from '../../../../../src/core/server';
 import { checkLicense } from '../check_license';
 
 export function registerLicenseChecker(server, pluginId, pluginName, minimumLicenseRequired) {
   const xpackMainPlugin = server.plugins.xpack_main;
-  const thisPlugin = server.plugins[pluginId];
-
-  mirrorPluginStatus(xpackMainPlugin, thisPlugin);
-  xpackMainPlugin.status.once('green', () => {
-    // Register a function that is called whenever the xpack info changes,
-    // to re-compute the license check results for this plugin
-    xpackMainPlugin.info
-      .feature(pluginId)
-      .registerLicenseCheckResultsGenerator((xpackLicenseInfo) => {
-        return checkLicense(pluginName, minimumLicenseRequired, xpackLicenseInfo);
-      });
-  });
+  server.newPlatform.setup.core.status.core$
+    .pipe(pairwise())
+    .subscribe(([coreLast, coreCurrent]) => {
+      if (
+        coreLast.elasticsearch.level !== ServiceStatusLevels.available &&
+        coreCurrent.elasticsearch.level === ServiceStatusLevels.available
+      ) {
+        // Register a function that is called whenever the xpack info changes,
+        // to re-compute the license check results for this plugin
+        xpackMainPlugin.info
```

Comment:
minor: it can be called more than once, it's okay as we are about to delete this code.

Reply:
Went ahead and added some unsubscribe logic to ensure this doesn't get called more than once